packages/dart/noports_core/lib/src/commands/activate/activate_impl.dart

```diff
@@ -0,0 +1,146 @@
+import 'dart:io';
+
+import 'package:at_auth/at_auth.dart';
+import 'package:at_client/at_client.dart';
+import 'package:at_onboarding_cli/at_onboarding_cli.dart';
+import 'package:at_utils/at_logger.dart';
+
+import 'package:noports_core/commands.dart';
+import 'package:noports_core/src/commands/activate/activate_params.dart';
+import 'package:noports_core/utils.dart';
+
+class Activate {
+  final ActivateParams _params;
+
+  final AtOnboardingService _onboardingService;
+
+  final logger = AtSignLogger('Activate', loggingHandler: CLILoggingHandler())
+    ..level = 'info';
+
+  Activate._(this._onboardingService, this._params);
```

Comment:
Nit: the strongly constructor shouldn't be private, most non-binary consumers of this api will prefer this over the factory constructor.

Reply:
Constructor made public through [90a00f6](https://github.com/atsign-foundation/noports/pull/2304/commits/90a00f64245a572d0c4a5daca989208fb03e2ec9)